warehouse/manage/views.py

```diff
@@ -1443,66 +1455,149 @@ def manage_project_roles(project, request, _form_class=CreateRoleForm):
+                project_name=project.name,
+                email_token=token,
+                token_age=token_service.max_age,
+            )
             project.record_event(
-                tag="project:role:add",
+                tag="project:role:invite",
                 ip_address=request.remote_addr,
                 additional={
                     "submitted_by": request.user.username,
                     "role_name": role_name,
                     "target_user": username,
                 },
             )
+            request.db.flush()  # in order to get id
+            request.session.flash(f"Invitation sent to '{username}'", queue="success")
 
-            owner_roles = (
-                request.db.query(Role)
-                .join(Role.user)
-                .filter(Role.role_name == "Owner", Role.project == project)
-            )
-            owner_users = {owner.user for owner in owner_roles}
+        form = _form_class(user_service=user_service)
 
-            # Don't send to the owner that added the new role
-            owner_users.discard(request.user)
+    roles = set(request.db.query(Role).join(User).filter(Role.project == project).all())
+    invitations = set(
+        request.db.query(RoleInvitation)
+        .join(User)
+        .filter(RoleInvitation.project == project)
+        .all()
+    )
 
-            # Don't send owners email to new user if they are now an owner
-            owner_users.discard(user)
+    return {
+        "project": project,
+        "roles": roles,
+        "invitations": invitations,
+        "form": form,
+    }
 
-            send_collaborator_added_email(
-                request,
-                owner_users,
-                user=user,
-                submitter=request.user,
-                project_name=project.name,
-                role=form.role_name.data,
-            )
 
-            send_added_as_collaborator_email(
-                request,
-                user,
-                submitter=request.user,
-                project_name=project.name,
-                role=form.role_name.data,
-            )
+@view_config(
+    route_name="manage.project.revoke_invite",
+    context=Project,
+    uses_session=True,
+    require_methods=["POST"],
+    permission="manage:project",
+    has_translations=True,
+)
+def revoke_project_role_invitation(project, request, _form_class=ChangeRoleForm):
+    user_service = request.find_service(IUserService, context=None)
+    token_service = request.find_service(ITokenService, name="email")
 
-            request.session.flash(
-                f"Added collaborator '{form.username.data}'", queue="success"
+    try:
+        user = user_service.get_user(request.POST["user_id"])
+        user_invite = (
+            request.db.query(RoleInvitation)
+            .filter(RoleInvitation.project == project)
+            .filter(RoleInvitation.user == user)
+            .one()
+        )
```

Comment:
I think we need to use the `RoleInvitation` ID here instead.

Reply:
As discussed, we decided to leave a minimal footprint in the role invitation table, thus marking this as resolved.